techlibs/quicklogic/synth_quicklogic.cc

```diff
@@ -0,0 +1,266 @@
+                top_opt = "-top " + args[++argidx];
+                continue;
+            }
+            if (args[argidx] == "-edif" && argidx+1 < args.size()) {
+                edif_file = args[++argidx];
+                continue;
+            }
+
+            if (args[argidx] == "-family" && argidx+1 < args.size()) {
+                family = args[++argidx];
+                continue;
+            }
+            if (args[argidx] == "-blif" && argidx+1 < args.size()) {
+                blif_file = args[++argidx];
+                continue;
+            }
+            if (args[argidx] == "-verilog" && argidx+1 < args.size()) {
+                verilog_file = args[++argidx];
+                continue;
+            }
+            if (args[argidx] == "-no_adder") {
+                inferAdder = false;
+                continue;
+            }
+            if (args[argidx] == "-no_abc_opt") {
+                abcOpt = false;
+                continue;
+            }
+            
+            break;
+        }
+        extra_args(args, argidx, design);
+
+        if (!design->full_selection())
+            log_cmd_error("This command only operates on fully selected designs!\n");
+
+        log_header(design, "Executing SYNTH_QUICKLOGIC pass.\n");
+        log_push();
+
+        run_script(design, run_from, run_to);
+
+        log_pop();
+    }
+
+    void script() override
+    {
+        if (check_label("begin")) {
+            std::string readVelArgs;
+            readVelArgs = " +/quicklogic/" + family + "_cells_sim.v";
+            
+            run("read_verilog -lib -specify +/quicklogic/cells_sim.v" + readVelArgs);
+            run(stringf("hierarchy -check %s", help_mode ? "-top <top>" : top_opt.c_str()));
+        }
+
+        if (check_label("prepare"))  {
+            run("proc");
+            run("flatten");
+            run("opt_expr");
+            run("opt_clean");
+            run("deminout");
+            run("opt");
+        }
+
+        if (check_label("coarse")) {
+            run("opt_expr");
+            run("opt_clean");
+            run("check");
+            run("opt");
+            run("wreduce -keepdc");
+            run("peepopt");
+            run("pmuxtree");
+            run("opt_clean");
+
+            run("alumacc");
+            run("opt");
+            run("fsm");
+            run("opt -fast");
+            run("memory -nomap");
+            run("opt_clean");
+        }
+
+        if (check_label("map_ffram")) {
+            run("opt -fast -mux_undef -undriven -fine");
+            run("memory_map -iattr -attr !ram_block -attr !rom_block -attr logic_block "
+                    "-attr syn_ramstyle=auto -attr syn_ramstyle=registers "
+                    "-attr syn_romstyle=auto -attr syn_romstyle=logic");
+            run("opt -undriven -fine");
+        }
+
+        if (check_label("map_gates")) {
+            if(family == "qlf_k4n8") {
+                run("async2sync");
```

Comment:
I don't think we should use this in the synthesis - this changes circuit behavior

Reply:
Agree, on this one anyways, I am thinking of removing sync2sync as are adding local reset in the arch and infer DFF instead latch in the netlist. I will remove it.